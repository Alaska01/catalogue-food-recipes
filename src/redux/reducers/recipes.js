const initialState = { food_recipes: [], food_item: {} };

const recipes = (state = initialState, action) => {
  switch (action.type) {
    case 'GET RECIPES':
      return { ...state, food_recipes: action.payload };
    case 'GET RECIPE':
      return { ...state, food_item: action.payload };
    case 'FILTER CATEGORY':
      return { ...state, food_recipes: action.payload.data.meals };
    case 'RESET_DATA':
      return { ...state, food_recipes: [] };
    default:
      return state;
  }
};

export default recipes;
