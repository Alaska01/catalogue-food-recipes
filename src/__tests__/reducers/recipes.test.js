import recipes from '../../reducers/recipes';

describe('recipes reducer', () => {
  it('should return the initial state', () => {
    expect(recipes(undefined, {})).toMatchSnapshot();
  });

  it('should handle Get RECIPES', () => {
    expect(recipes([], {
      type: 'GET RECIPES',
    })).toMatchSnapshot();
  });

  it('should handle GET RECIPE', () => {
    expect(recipes({}, {
      type: 'GET RECIPE',
    })).toMatchSnapshot();
  });

  it('should handle RESET DATA', () => {
    expect(recipes([], {
      type: 'RESET_DATA',
    })).toMatchSnapshot();
  });
});
