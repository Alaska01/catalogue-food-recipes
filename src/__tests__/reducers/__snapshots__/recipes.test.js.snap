// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipes reducer should handle GET RECIPE 1`] = `
Object {
  "food_item": undefined,
}
`;

exports[`recipes reducer should handle Get RECIPES 1`] = `
Object {
  "food_recipes": undefined,
}
`;

exports[`recipes reducer should handle RESET DATA 1`] = `
Object {
  "food_recipes": Array [],
}
`;

exports[`recipes reducer should return the initial state 1`] = `
Object {
  "food_item": Object {},
  "food_recipes": Array [],
}
`;
